libs/api/src/lib/media/novaMediaAdapter.ts

```diff
@@ -0,0 +1,148 @@
+import {URL} from 'url'
+import FormData from 'form-data'
+import {ImageWithFocalPoint} from '../db/image'
+import fetch from 'node-fetch'
+import type {FileUpload} from 'graphql-upload'
+import {
+  ArrayBufferUpload,
+  MediaAdapter,
+  UploadImage,
+  ImageTransformation,
+  ImageRotation
+} from '@wepublish/image/api'
+import {MediaServerError} from './karmaMediaAdapter'
+
+export class NovaMediaAdapter implements MediaAdapter {
+  readonly url: URL
+  readonly token: string
+  readonly internalURL: URL
+
+  constructor(url: URL, token: string, internalURL: URL = url) {
+    this.url = url
+    this.token = token
+    this.internalURL = internalURL
+  }
+
+  async _uploadImage(form: FormData): Promise<UploadImage> {
+    // The form-data module reports a known length for the stream returned by createReadStream,
+    // which is wrong, override it and always set it to false.
+    // Related issue: https://github.com/form-data/form-data/issues/394
+    form.hasKnownLength = () => false
+
+    const response = await fetch(this.internalURL, {
+      method: 'POST',
+      headers: {authorization: `Bearer ${this.token}`},
+      body: form
+    })
+
+    const json = await response.json()
```

Comment:
maybe some other more descriptive name would be better? like parsed, responseAsJson, parsedResponse or something? it would help to understand better on line 44 as well

Reply:
Was just c&p from the old one haha :D Can change that and clean that up yeah. Didn't really clean anything up yet in general